packages/amazonq/src/lsp/utils.ts

```diff
@@ -0,0 +1,20 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as vscode from 'vscode'
+
+export function getCursorState(selection: readonly vscode.Selection[]) {
```

Comment:
this could live in textDocumentUtils

Reply:
This function is pretty specific to the amazonQ LSP as its converting from `vscode.Selection[]` type to the `CursorState` type the LSP is expecting. I added a comment and type signature to make this more clear.